### PR TITLE
Add option to use CATCHUP_SKIP_KNOWN_RESULTS_FOR_TESTING in PubnetParallelCatchupV2

### PR DESCRIPTION
**Pull request by ThomasBrady**
# Overview
* Adds `--catchup-use-known-results-for-testing` cli flag to supercluster, which results `PubnetParallelCatchupV2` running with `CATCHUP_SKIP_KNOWN_RESULTS_FOR_TESTING=true` in its config
* Adds `worker.catchup_skip_known_results_for_testing` kube set option, which is set to true if supercluster was run with `--catchup-use-known-results-for-testing`
* The catchup workers `config` volume is sourced from `stellar-core-config` configMap
*  if `worker.use_known_results_config` is true, the data of the `stellar-core-config` `configMap` includes `CATCHUP_SKIP_KNOWN_RESULTS_FOR_TESTING = true`

# Testing:
- Currently, the [catchup worker invocation](https://github.com/stellar/supercluster/compare/main...ThomasBrady:supercluster:add-catchup-skip-known-results-missionops?expand=1#diff-8ad1481eef0b22a40c79c91cf31a8a6866ca404224797fab02378208636f9799R34) is set to `--ll=INFO` to allow for log message which validate results are downloaded and used for catchup 
- Locally I have run `PubnetParallelCatchupV2` via helm and been able to see the `--use_known_results_config` flag resulting in Results being downloaded via logs
```
stellar_core_image: docker-registry.services.stellar-ops.com/dev/stellar-core:22.0.1-2178.fb725ab9b.focal-do-not-use-in-prd-buildtests
  use_known_results_config: true
  ```
  - Next step is to test via the [supercluster test pipeline](https://buildmeister-v3.stellar-ops.com/job/Core/job/stellar-supercluster-test/) with [my changes to `stellar-supercluster`](https://github.com/stellar/stellar-supercluster/pull/323)
  
  # Future
  It would be nice to have a dynamic (or templated) generation of config files for this mission. We use a TOML library in the other missions to generate config files, perhaps we can do something similar for `PubnetParallelCatchupV2`?

### CODE REVIEW

File: src/FSLibrary/MissionHistoryPubnetParallelCatchupV2.fs [hunk 1]
```diff
@@ -21,7 +21,7 @@ open k8s
 
 // Constants
 let helmReleaseName = "parallel-catchup"
-let helmChartPath = "/supercluster/src/MissionParallelCatchup/parallel_catchup_helm"
+let helmChartPath = "src/MissionParallelCatchup/parallel_catchup_helm"
```
Comment (ThomasBrady) [inline src/FSLibrary/MissionHistoryPubnetParallelCatchupV2.fs:24]: For local testing.

File: src/App/Program.fs [hunk 2]
```diff
@@ -458,6 +459,12 @@ type MissionOptions
              Required = false)>]
     member self.NumRuns = numRuns
 
+    [<Option("catchup-skip-known-results-for-testing",
+             HelpText = "skip known results for testing",
```
Comment (jayz22) [inline src/App/Program.fs:463]: don't forget to put a more specific help text

File: src/App/Program.fs [hunk 3]
```diff
@@ -19,7 +19,7 @@ open System.Threading
 
 
 type KubeOption(kubeConfig: string, namespaceProperty: string option) =
-    [<Option('k', "kubeconfig", HelpText = "Kubernetes config file", Required = false, Default = "~/.kube/config")>]
+    [<Option('k', "kubeconfig", HelpText = "Kubernetes config file test extra strings in help to see if correct supercluster is running", Required = false, Default = "~/.kube/config")>]
```
Comment (jayz22) [inline src/App/Program.fs:22]: should this be removed as well after testing?

File: src/MissionParallelCatchup/parallel_catchup_helm/files/worker.sh [hunk 4]
```diff
@@ -24,14 +24,16 @@ while true; do
 # Fetch the next job key from the Redis queue. 
 # Our ranges are generated in the order we want to run them from left to right, so we always pull from the left
 JOB_KEY=$(redis-cli -h "$REDIS_HOST" -p "$REDIS_PORT" LMOVE "$JOB_QUEUE" "$PROGRESS_QUEUE" LEFT LEFT)
-
+echo "BEGIN CONFIG"
+cat  /config/stellar-core.cfg 
+echo "END CONFIG"
```
Comment (jayz22) [inline src/MissionParallelCatchup/parallel_catchup_helm/files/worker.sh:29]: please remove after testing

File: src/MissionParallelCatchup/parallel_catchup_helm/files/worker.sh [hunk 5]
```diff
@@ -24,14 +24,16 @@ while true; do
...
 if [ -n "$JOB_KEY" ]; then
     # Start timer
     START_TIME=$(date +%s)  
     echo "Processing job: $JOB_KEY"
 
-    if [ ! "$(/usr/bin/stellar-core --conf /config/stellar-core.cfg new-db --console&&
-    /usr/bin/stellar-core --conf /config/stellar-core.cfg catchup "$JOB_KEY" --metric 'ledger.transaction.apply' --console)" ]; then
+    if [ ! "$(/usr/bin/stellar-core --ll=INFO --conf /config/stellar-core.cfg new-db --console&&
+    /usr/bin/stellar-core --ll=INFO --conf /config/stellar-core.cfg catchup "$JOB_KEY" --metric 'ledger.transaction.apply' --console)" ]; then
```
Comment (jayz22) [inline src/MissionParallelCatchup/parallel_catchup_helm/files/worker.sh:36]: please remove after testing

**Review by jayz22 — COMMENTED** [review]
This approach is okay for now. But I think after the experimental phase is done we should properly move this option to stellar core command line. 
Conditionally appending options to the configmap can get out of hand quickly and be hard to maintain. 
Let's not forget to create a followup issue for that.

**Comment by ThomasBrady** [comment]
I was able to run with this option enabled in the `test-stellar-supercluster` pipeline https://buildmeister-v3.stellar-ops.com/job/Core/job/stellar-supercluster-test/81/

**Review by jayz22 — APPROVED** [review]

**Comment by jayz22** [comment]
> I was able to run with this option enabled in the `test-stellar-supercluster` pipeline https://buildmeister-v3.stellar-ops.com/job/Core/job/stellar-supercluster-test/81/

Looks good, do you want to switch the default flag to false as @anupsdf suggested, or do that on the Jenkins side?

**Comment by ThomasBrady** [comment]
> > I was able to run with this option enabled in the `test-stellar-supercluster` pipeline https://buildmeister-v3.stellar-ops.com/job/Core/job/stellar-supercluster-test/81/
> 
> Looks good, do you want to switch the default flag to false as @anupsdf suggested, or do that on the Jenkins side?

Thanks. I just pushed a change to make this default to `true`. I also have [this PR ](https://github.com/stellar/stellar-supercluster/pull/323)to update `stellar-supercluster`.

Do you have permissions to edit the Jenkins schedule to ensure we do a once weekly non-skipping catchup run?

**Review by jayz22 — COMMENTED** [review]
> > > I was able to run with this option enabled in the `test-stellar-supercluster` pipeline https://buildmeister-v3.stellar-ops.com/job/Core/job/stellar-supercluster-test/81/
> > 
> > 
> > Looks good, do you want to switch the default flag to false as @anupsdf suggested, or do that on the Jenkins side?
> 
> Thanks. I just pushed a change to make this default to `true`. I also have [this PR ](https://github.com/stellar/stellar-supercluster/pull/323)to update `stellar-supercluster`.
> 
> Do you have permissions to edit the Jenkins schedule to ensure we do a once weekly non-skipping catchup run?

No you need to set it up. There is also the policy question of when and how we use this mode (which is separate from this PR). I'm not sure if we have a time-based CI job currently. 

Just want to double check, are we sure about setting default to `true`. Or should we instead have this option be opted-in, i.e. configured from outside. (which is what @MonsieurNicolas has [suggested](https://github.com/stellar/stellar-core/issues/2814#issue-747861077) at the beginning). Since that issue is closed I suppose all the work mentioned has been completed and it is okay to turn this on by default?

**Comment by jayz22** [comment]
> The linked issue is referencing `stellar-core catchup`. This mode is opt-in in `stellar-core` (in fact one can only opt-in when running catchup on a test build). As discussed with Anup in standup this morning we want all supercluster runs to use this mode by default, with a less frequent weekly non-skip ParallelCatchupV2 run. Based on that, I think it makes sense to default this mode being enabled in our supercluster runs.

Sounds good, thanks

**Pull request merged**